tools/dtest/config/config.go

```diff
@@ -0,0 +1,130 @@
+package config
+
+import (
+	"fmt"
+	"time"
+
+	m3dbnode "github.com/m3db/m3db/x/m3em/node"
+
+	etcdclient "github.com/m3db/m3cluster/client/etcd"
+	"github.com/m3db/m3cluster/services"
+	"github.com/m3db/m3cluster/services/placement"
+	"github.com/m3db/m3em/generated/proto/m3em"
+	"github.com/m3db/m3em/node"
+	"github.com/m3db/m3x/config"
+	xlog "github.com/m3db/m3x/log"
+	"google.golang.org/grpc"
+)
+
+var (
+	defaultConnectionTimeout = 2 * time.Minute
+)
+
+// Configuration is a collection of knobs to control test behavior
+type Configuration struct {
+	DTest DTestConfig              `yaml:"dtest"`
+	M3EM  M3EMConfig               `yaml:"m3em"`
+	KV    etcdclient.Configuration `yaml:"kv"`
+}
+
+// TODO(prateek): merge DTest/M3EM
+// TODO(prateek): make config structs for options in m3em
+
+// M3EMConfig is a list of m3em environment settings
+type M3EMConfig struct {
+	HeartbeatPort int                 `yaml:"heartbeatPort" validate:"nonzero"`
+	AgentPort     int                 `yaml:"agentPort" validate:"nonzero"`
+	M3DBPort      int                 `yaml:"m3dbPort" validate:"nonzero"`
+	M3DBServiceID string              `yaml:"m3dbServiceID" validate:"nonzero"`
+	Instances     []PlacementInstance `yaml:"instances" validate:"min=1"`
+	Node          node.Configuration  `yaml:"node"`
+}
+
+// DTestConfig is a collection of DTest configs
+type DTestConfig struct {
+	DebugPort            int `yaml:"debugPort" validate:"nonzero"`
+	NumShards            int `yaml:"numShards" validate:"nonzero"`
+	BootstrapTimeoutMins int `yaml:"bootstrapTimeoutMins" validate:"nonzero"`
+}
+
+// PlacementInstance is a config for a services.PlacementInstance
+type PlacementInstance struct {
+	ID       string `yaml:"id" validate:"nonzero"`
+	Rack     string `yaml:"rack" validate:"nonzero"`
+	Zone     string `yaml:"zone" validate:"nonzero"`
+	Weight   uint32 `yaml:"weight" validate:"nonzero"`
+	Hostname string `yaml:"hostname" validate:"nonzero"`
+}
+
+// New constructs a Configuration object from the path specified
+func New(m3emConfigPath string) (*Configuration, error) {
+	var conf Configuration
+	if err := xconfig.LoadFile(&conf, m3emConfigPath); err != nil {
+		return nil, err
+	}
+
+	return &conf, nil
+}
+
+// M3DBNodes returns a slice of m3dbnode.Nodes per the config provided
+func (mc *M3EMConfig) M3DBNodes(opts node.Options, numNodes int) ([]m3dbnode.Node, error) {
```

Comment:
Maybe we should just call this `Nodes()` and let the types speak for themselves.